test-data/unit/check-attr.test

```diff
@@ -771,3 +771,29 @@ class FFrozen(F):
     def bar(self) -> bool:
         return self._cb(5, 6)
 [builtins fixtures/callable.pyi]
+
+[case testAttrsWithFactory]
+from typing import List
+import attr
+def my_factory() -> int:
+    return 7
+@attr.s
+class A:
+    x: List[int] = attr.ib(factory=list)
+    y: int = attr.ib(factory=my_factory)
+A()
+[builtins fixtures/list.pyi]
+
+[case testAttrsFactoryAndDefault]
+import attr
+@attr.s
+class A:
+    x: int = attr.ib(factory=int, default=7)  # E: Can't pass both `default` and `factory`.
+[builtins fixtures/bool.pyi]
+
+[case testAttrsFactoryBadReturn]
+import attr
+@attr.s
+class A:
+    x: int = attr.ib(factory=list)  # E: Incompatible types in assignment (expression has type "List[T]", variable has type "int")
```

Comment:
The error message looks a bit cryptic. Why it is `List[T]` and not `List[Any]`? There is no `T` in code. I suppose this might be a bug in using class as factory, is the error better if you use a generic function instead of class?

Reply:
This error come from the stubs not from the plugin (i.e. any method that takes a type would give this error message.)  I can add a generic function too.